### PR TITLE
Add metrics to pinga jobs

### PR DESCRIPTION
**Pull request by paulocsanz**
Improves telemetry/metrics of `execute_job_task`.

Logs the number of concurrent tasks and limit every time a new job is spawned in pinga. Also logs if pinga is at full capacity, which means no new tasks will be consumed until one is finished.

We also are logging the status of execution, jobs without status are still being executed, so we can easily keep track of what is going on at each instance.

### CODE REVIEW

**Comment by paulocsanz** [comment]
bors merge

File: lib/pinga-server/src/server.rs [hunk 1]
```diff
@@ -419,6 +438,8 @@ async fn execute_job_task(
     metadata: Arc<ServerMetadata>,
```
Comment (britmyerss) [inline lib/pinga-server/src/server.rs:438]: it's the lines write above this, but if we add telemetry to this func, and record the fields you added, we'd be able to do some fancy graphing in honeycomb to show how these numbers change over time!

File: lib/pinga-server/src/server.rs [hunk 2]
```diff
@@ -361,17 +366,26 @@ async fn receive_job_requests(
     Ok(())
 }
 
+const CONCURRENT_TASKS: AtomicUsize = AtomicUsize::new(0);
+
```
Comment (britmyerss) [inline lib/pinga-server/src/server.rs:370]: same thing here! Something like 
``` #[instrument (level="info", skip(rx)]```

File: lib/pinga-server/src/server.rs (hunk 1 above)
Comment (paulocsanz) [reply lib/pinga-server/src/server.rs:438]: Sorry, not sure I understand

File: lib/pinga-server/src/server.rs (hunk 2 above)
Comment (paulocsanz) [reply lib/pinga-server/src/server.rs:370]: Why would we want a span for the task that spawns task?

Comment (paulocsanz) [reply lib/pinga-server/src/server.rs:370]: Something like this?

image

Comment (britmyerss) [reply lib/pinga-server/src/server.rs:370]: yep!!

File: lib/pinga-server/src/server.rs (hunk 1 above)
Comment (britmyerss) [reply lib/pinga-server/src/server.rs:438]: oh sorry - this IS in the instrument tag. ignore me!

**Review by nickgerace — APPROVED** [review]
:rocket:

File: lib/pinga-server/src/server.rs (hunk 2 above)
Comment (paulocsanz) [reply lib/pinga-server/src/server.rs:370]: Done!

**Comment by paulocsanz** [comment]
bors merge

**Comment by paulocsanz** [comment]
bors merge

**Comment by paulocsanz** [comment]
bors merge

**Pull request merged**